juanevillam/xtrovert
# Configure ESLint and Prettier for Code Quality

## Overview  
This PR sets up ESLint and Prettier for the project to enforce code quality and consistent formatting standards.

---

## Key Changes  

### **Features**  
- Installed ESLint and Prettier packages.  
- Added `.vscode/settings.json` for editor configurations.  
- Created and configured `.eslintrc.json` with appropriate rules and plugins.  
- Added `.prettierignore` and `.prettierrc` files for Prettier settings.  

### **Enhancements**  
- Updated `package.json` with new scripts:  
  - `"lint": "next lint"`  
  - `"lint:fix": "eslint . --ext .js,.jsx,.ts,.tsx --fix"`  
  - `"format": "prettier --write '**/*.{js,jsx,ts,tsx,json,css,md}'"`  
- Verified linting and formatting commands to ensure they work as expected.  

---

## Reasons for Changes  
Introducing ESLint and Prettier ensures that all code adheres to consistent style and quality guidelines, improving code readability and maintainability.

---

## How to Test  
1. Run `npm run lint` to check for linting errors.  
2. Run `npm run lint:fix` to automatically fix linting issues.  
3. Run `npm run format` to format the code according to Prettier rules.  
4. Verify that the project builds and runs without issues after formatting.
